app/models/product.rb

```diff
@@ -34,4 +36,12 @@ def ensure_not_referenced_by_any_line_item
         throw :abort
       end
     end
+
+    def give_default_title
+      self.title='abc'
```

Comment:
why have we used self here? please explain.

Reply:
self.title is used instead of just title because title is not a local variable, it's an attribute of the Product model.So using self.title is a way to ensure that we are accessing the title attribute of the Product instance, and not creating a new local variable with the same name.